pandas/tests/test_panel.py

```diff
@@ -469,7 +469,13 @@ def test_delitem_and_pop(self):
 
     def test_setitem(self):
         lp = self.panel.filter(['ItemA', 'ItemB']).to_frame()
-        with pytest.raises(ValueError):
+
+        # On master we go all the way down to
+        # MultiIndex.from_tuples(DatetimeIndex), which raise a
+        # ValueError: cannot include dtype 'M' in a buffer
+        # Now we (correctly) raise a TypeError.
+        # TODO: Add release note for this.
```

Comment:
How much do we care about compatibility here?


```python
In [2]: pd.MultiIndex.from_tuples(pd.date_range('2000', periods=4))
```

On master, we raise with

```pytb
-> 2253 def tuples_to_object_array(ndarray[object] tuples):
   2254     cdef:
   2255         Py_ssize_t i, j, n, k, tmp

ValueError: cannot include dtype 'M' in a buffer
```


now we (well, Cython) raise with

```pytb
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-bf92ca748af3> in <module>
----> 1 pd.MultiIndex.from_tuples(pd.date_range('2000', periods=4))

~/sandbox/pandas/pandas/core/indexes/multi.py in from_tuples(cls, tuples, sortorder, names)
    378                 tuples = tuples._values
    379
--> 380             arrays = list(lib.tuples_to_object_array(tuples).T)
    381         elif isinstance(tuples, list):
    382             arrays = list(lib.to_object_array_tuples(tuples).T)

TypeError: Argument 'tuples' has incorrect type (expected numpy.ndarray, got DatetimeArrayMixin)
```